media/js/firefox/welcome/welcome8.js

```diff
@@ -2,9 +2,72 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-(function() {
+(function (Mozilla) {
     'use strict';
 
+    /* update dataLayer with experiment info */
+    var href = window.location.href;
+
+    var initTrafficCop = function () {
+        if (href.indexOf('v=') !== -1) {
+            var fbContainerLink = document.getElementById('fb-container-link');
+            var monitorLink =  document.getElementById('monitor-link');
+            var utmSource = '';
+            if (href.indexOf('v=text') !== -1) {
+                utmSource = 'mozilla.org-firefox-welcome-8-text';
+                // default, no special class needed
+                window.dataLayer.push({
+                    'data-ex-variant': 'experiment-hvt-visual-text',
+                    'data-ex-name': 'experiment-hvt-visual'
+                });
+            } else if (href.indexOf('v=image') !== -1) {
+                utmSource = 'mozilla.org-firefox-welcome-8-image';
+                document.body.classList.add('variation-2-image');
```

Comment:
Because this experiment logic should be in the `<head>` of the page for performance reasons, it means you might need to move logic like this into the Python view / Jinja template.